src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -1467,24 +1470,24 @@ App::DocumentObjectExecReturn *Hole::execute(void)
             const double pi = 3.14159265359;
             bool leftHanded = (bool) ThreadDirection.getValue();
 
-            // Nomenclature and formulae according to Schaeffler Technical pocket guide (edition 2018)
-            // this is the same for all metric and UTS threads.
-            double D = threadDescription[ThreadType.getValue()][ThreadSize.getValue()].diameter;
-            double P = ThreadPitch.getValue();
-            double H1 = 0.54127 * P;
-            double H = 0.86603 * P;
-            double D1 = D - 2 * H1;
-            double alfa = 60.0 * pi/180;
-
-
-            double clearance;   // clearance used for Actual Thread modeling.
+            // Nomenclature and formulae according to Figure 1 of ISO 68-1
+            // this is the same for all metric and UTS threads as stated here:
+            // https://en.wikipedia.org/wiki/File:ISO_and_UTS_Thread_Dimensions.svg
+            // Note that in the ISO standard, Dmaj is called D, which has been followed here.
+            double D = threadDescription[ThreadType.getValue()][ThreadSize.getValue()].diameter;  //
+            double P = ThreadPitch.getValue();            // Pitch
+            double H = sqrt(3.0)/2.0 * P;                 // Height of fundamentral triangle
+            double H1 = 5*H/8 * P;                        // The thread depth
+            double D1 = D - 2 * H1;                       // Basic minor diameter of internal thread
+            double alfa = Base::toRadians(60.0);          // Angle of the fundamental triangle. Same for both ISO and UTS.
+
+            double clearance;                             // clearance to be added on the diameter
             if (UseCustomThreadClearance.getValue())
                 clearance = CustomThreadClearance.getValue();
             else
                 clearance = getThreadClassClearance();
 
-
-            // construct the cross section
+            // construct the cross section going counter-clockwise
             gp_Pnt p1 = toPnt(xDir * D1/2.0 + P*zDir + clearance/2.0*xDir);

```

Comment:
I insist to describe this better. It is worth the effort because once we have an issue and look into the code, we will safe a lot of time. Thus please add each a line before the calculation describing what you do and refer to https://en.wikipedia.org/wiki/File:ISO_and_UTS_Thread_Dimensions.svg
Then it should become clear how you move along the path defined by the norm point by point.

Reply:
Alright :)

The original code was too general in the sense that it supported arbitrary angle (not just  60 deg). Since I since found out that UTS and ISO use the same angle this is no longer necessary. I have also included a reference to a forum post where I give a graphical explanation of the construction with reference to the ISO standard. 

Please have a look

Reply:
https://forum.freecadweb.org/viewtopic.php?f=19&t=54284#p466570

Reply:
> Please have a look

many thanks. Now the code looks good from my point of view. I will compile and test later today.